### PR TITLE
[add tencentcloud_sqlserver_db resource]

### PR DESCRIPTION
**Pull request by brickzzhang**
1. add tencentcloud_sqlserver_db resource
2. example contents depend on sqlserver instance

[commiter: brickzzhang]

-------

➜  terraform-provider-tencentcloud git:(master) make testacc TESTARGS=" -run=TestAccTencentCloudSqlserverDB_basic_and_update"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccTencentCloudSqlserverDB_basic_and_update -timeout 120m
?   	github.com/terraform-providers/terraform-provider-tencentcloud	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/terraform-providers/terraform-provider-tencentcloud/gendoc	(cached) [no tests to run]
=== RUN   TestAccTencentCloudSqlserverDB_basic_and_update
--- PASS: TestAccTencentCloudSqlserverDB_basic_and_update (241.94s)
PASS
ok  	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud	242.744s
?   	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/connectivity	[no test files]
?   	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/internal/helper	[no test files]
?   	github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/ratelimit	[no test files]

### CODE REVIEW

File: examples/tencentcloud-sqlserver/main.tf [hunk 1]
```diff
@@ -1,2 +1,9 @@
 data "tencentcloud_sqlserver_zone_config" "mysqlserver" {
 }
+
+resource "tencentcloud_sqlserver_db" "mysqlserver_db" {
+  instance_id = "mssql-3cdq7kx5"
```
Comment (gailwang) [inline examples/tencentcloud-sqlserver/main.tf]: 写example的时候用tf里创建的example instance，否则客户拿到跑不起来

File: tencentcloud/resource_tc_sqlserver_db.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+	}
+
+	dbName := d.Get("name").(string)
+	charset := d.Get("charset").(string)
+	remark := d.Get("remark").(string)
+
+	if err := sqlserverService.CreateSqlserverDB(ctx, instanceID, dbName, charset, remark); err != nil {
+		return err
+	}
+
```
Comment (gailwang) [inline tencentcloud/resource_tc_sqlserver_db.go:100]: create里没有写SetId吗

File: tencentcloud/resource_tc_sqlserver_db.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+	if !has {
+		d.SetId("")
+		return nil
+	}
+	_ = d.Set("charset", *dbInfo.Charset)
+	_ = d.Set("remark", *dbInfo.Remark)
+	_ = d.Set("create_time", *dbInfo.CreateTime)
+	_ = d.Set("status", SQLSERVER_DB_STATUS[*dbInfo.Status])
+
+	d.SetId(name)
```
Comment (gailwang) [inline tencentcloud/resource_tc_sqlserver_db.go]: 为什么在Read里SetID？

File: tencentcloud/resource_tc_sqlserver_db.go [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+func resourceTencentCloudSqlserverDBUpdate(d *schema.ResourceData, meta interface{}) error {
+	defer logElapsed("resource.tencentcloud_sqlserver_db.update")()
+
+	logId := getLogId(contextNil)
+	ctx := context.WithValue(context.TODO(), logIdKey, logId)
+
+	sqlserverService := SqlserverService{client: meta.(*TencentCloudClient).apiV3Conn}
+	instanceId := d.Get("instance_id").(string)
+
```
Comment (gailwang) [inline tencentcloud/resource_tc_sqlserver_db.go:144]: 如果是分开设置更新，需要用到
d.Partial(true)
//update1 ...
//update2 ...
d.Partial(false)

File: tencentcloud/resource_tc_sqlserver_db.go [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+	if d.HasChange("name") {
+		oldValue, newValue := d.GetChange("name")
+		if err := sqlserverService.ModifySqlserverDBName(ctx, instanceId, oldValue.(string), newValue.(string)); err != nil {
+			return err
```
Comment (gailwang) [inline tencentcloud/resource_tc_sqlserver_db.go]: 如果name是可以修改的，为什么要拿他作为唯一ID？

File: tencentcloud/resource_tc_sqlserver_db.go [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+func resourceTencentCloudSqlserverDBDelete(d *schema.ResourceData, meta interface{}) error {
+	defer logElapsed("resource.tencentcloud_sqlserver_db.delete")()
+
+	logId := getLogId(contextNil)
+	ctx := context.WithValue(context.TODO(), logIdKey, logId)
+
+	sqlserverService := SqlserverService{client: meta.(*TencentCloudClient).apiV3Conn}
+	instanceId := d.Get("instance_id").(string)
+	name := d.Get("name").(string)
+
```
Comment (gailwang) [inline tencentcloud/resource_tc_sqlserver_db.go:163]: 有没有测试过，如果一个Instance不存在的情况下，去调用删除它的DB，会报什么错？会造成程序崩溃吗？

File: tencentcloud/service_tencentcloud_sqlserver.go [hunk 7]
```diff
@@ -78,3 +82,313 @@ func (me *SqlserverService) DescribeProductConfig(ctx context.Context, zone stri
...
+func (me *SqlserverService) ModifySqlserverDBRemark(ctx context.Context, instanceId string, dbname string, remark string) (errRet error) {
+	logId := getLogId(ctx)
+	request := sqlserver.NewModifyDBRemarkRequest()
+	request.InstanceId = &instanceId
+	var dbRemarkList []*sqlserver.DBRemark
+	var dbRemark sqlserver.DBRemark
+	dbRemark.Name = &dbname
+	dbRemark.Remark = &remark
+	dbRemarkList = append(dbRemarkList, &dbRemark)
+	request.DBRemarks = dbRemarkList
```
Comment (gailwang) [inline tencentcloud/service_tencentcloud_sqlserver.go]: request.DBRemarks = []*sqlserver.DBRemark{{Name:&dbName, Remark:&remark}}

File: tencentcloud/service_tencentcloud_sqlserver.go [hunk 8]
```diff
@@ -78,3 +82,313 @@ func (me *SqlserverService) DescribeProductConfig(ctx context.Context, zone stri
...
+	request.InstanceId = &instanceId
+	request.Names = []*string{&name}
+	defer func() {
+		if errRet != nil {
+			log.Printf("[CRITAL]%s api[%s] fail,reason[%s]", logId, request.GetAction(), errRet.Error())
+		}
+	}()
+
+	var response *sqlserver.DeleteDBResponse
+	err := resource.Retry(10*writeRetryTimeout, func() *resource.RetryError {
```
Comment (gailwang) [inline tencentcloud/service_tencentcloud_sqlserver.go]: 这里的10是测试过一定要用这么长的等待时间吗

File: tencentcloud/service_tencentcloud_sqlserver.go [hunk 9]
```diff
@@ -78,3 +82,313 @@ func (me *SqlserverService) DescribeProductConfig(ctx context.Context, zone stri
...
+		ratelimit.Check(request.GetAction())
+		result, e := me.client.UseSqlserverClient().DeleteDB(request)
+		if e != nil {
+			log.Printf("[CRITAL]%s DescribeDBs fail, reason:%s\n", logId, e.Error())
```
Comment (gailwang) [inline tencentcloud/service_tencentcloud_sqlserver.go]: Describe这里是不是有问题，这里的action用%s , request.GetAction 就不用来回改了

File: tencentcloud/resource_tc_sqlserver_db.go [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+	name := d.Get("name").(string)
+	dbInfo, has, err := sqlserverService.DescribeDBDetailsByName(ctx, instanceId, name)
+	if err != nil {
+		return err
+	}
+	if !has {
+		d.SetId("")
+		return nil
+	}
+	_ = d.Set("charset", *dbInfo.Charset)
```
Comment (gailwang) [inline tencentcloud/resource_tc_sqlserver_db.go]: 这里不用*

File: tencentcloud/resource_tc_sqlserver_db.go [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+			"create_time": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Database creation time.",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Database status. Valid values are `creating`, `running`, `modifying`, `dropping`.",
+			},
```
Comment (gailwang) [inline tencentcloud/resource_tc_sqlserver_db.go:71]: Valid value的意思是要输入的时候输入这些，现在这个参数是status，告诉客户分别代表什么含义

File: tencentcloud/resource_tc_sqlserver_db.go [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+		Schema: map[string]*schema.Schema{
+			"instance_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "SQL server instance ID which DB belongs to.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of DB. The DataBase name must be unique and must be composed of numbers, letters and underlines, and the first one can not be underline.",
```
Comment (gailwang) [inline tencentcloud/resource_tc_sqlserver_db.go]: Name of the SQL Server DB

**Comment by brickzzhang** [comment]
➜  terraform-provider-tencentcloud git:(master) ✗ make testacc TESTARGS=" -run=TestAccTencentCloudSqlserverDB_basic_and_update"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccTencentCloudSqlserverDB_basic_and_update -timeout 120m
?     github.com/terraform-providers/terraform-provider-tencentcloud  [no test files]
testing: warning: no tests to run
PASS
ok    github.com/terraform-providers/terraform-provider-tencentcloud/gendoc (cached) [no tests to run]
=== RUN   TestAccTencentCloudSqlserverDB_basic_and_update
--- PASS: TestAccTencentCloudSqlserverDB_basic_and_update (193.43s)
PASS
ok    github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud 194.689s
?     github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/connectivity  [no test files]
?     github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/internal/helper [no test files]
?     github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/ratelimit [no test files]

**Review by gailwang — APPROVED** [review]

**Pull request merged**